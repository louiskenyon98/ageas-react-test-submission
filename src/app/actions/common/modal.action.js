import {
    CLOSE_MODAL, SHOW_CONFIRMATION_MODAL,
    SHOW_STATUS_MODAL
} from '../types';


//Actions, not action creators.
export const showSuccessModal = (details) => ({
    type: SHOW_STATUS_MODAL,
    payload: {
        status: 'success',
        ...details
    }
});
//Object passed in (details), expecting body and title within object.
export const showFailModal = (details) => ({
    type: SHOW_STATUS_MODAL,
    payload: {
        status: 'fail',
        ...details
    }
});

export const showDeleteNoteConfirmationModal = (id) => {
    showConfirmationModal({
        title: 'Do you want to deletethisnote',
        accept: {
            label: 'YES',
            callback: () => deleteNote(id)
        },
        decline: {
            label: 'NO',
            callback: closeModal
        }
    })
};
export const showConfirmationModal = (details) => ({
    type: SHOW_CONFIRMATION_MODAL,
    payload: details
});

export const closeModal = () => ({
    type: CLOSE_MODAL,
});
