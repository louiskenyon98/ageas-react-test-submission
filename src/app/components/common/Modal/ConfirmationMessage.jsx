import React from 'react';
import styles from '../../../../themes/style.scss';

export default class ConfirmationMessage extends React.PureComponent {
    render() {
        return (
            <React.Fragment>
                <h1>Is you sure fam?</h1>
                {this.props.body}
                <button className={`${styles.modalButton} ui inverted primary button`}
                        onClick={this.props.accept.callback}>{this.props.accept.label}
                </button>
            </React.Fragment>
        )
    }
}
