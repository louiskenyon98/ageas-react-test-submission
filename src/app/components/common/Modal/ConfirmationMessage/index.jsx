import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../../../themes/style.scss';

export default class ConfirmationMessage extends React.PureComponent {
    render() {
        return (
            <React.Fragment>
                <h1>{this.props.title}</h1>
                {this.props.body}
                <div style={{float: 'right' , margin: 20}}>
                    {this.props.decline && <button
                        className={`${styles.modalButton} ui inverted primary button`}
                        // onClick={this.props.accept.callback}>{this.props.accept.label}
                        onClick={this.props.decline.onClick}
                    >
                        {this.props.decline.label}
                    </button>}
                    {this.props.accept && <button
                        className={`${styles.modalButton} ui inverted primary button`}
                        // onClick={this.props.accept.callback}>{this.props.accept.label}
                        onClick={this.props.accept.onClick}
                    >
                        {this.props.accept.label}
                    </button>}
                </div>
            </React.Fragment>
        )
    }
}

ConfirmationMessage.propTypes = {
    title: PropTypes.string.isRequired,
    body: PropTypes.string,
    decline: PropTypes.shape({
        label: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired
    }),
    accept: PropTypes.shape({
        label: PropTypes.string.isRequired,
        onClick: PropTypes.func.isRequired
    }),
};
