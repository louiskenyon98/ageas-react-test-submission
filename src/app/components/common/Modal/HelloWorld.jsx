import React from 'react';

export default class HelloWorld extends React.PureComponent {
    render() {
        return (
            <div>
                {this.props.message}
            </div>
        )
    }
}
