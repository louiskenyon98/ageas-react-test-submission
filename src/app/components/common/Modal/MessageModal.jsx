import React from 'react';
//:todo rename messagemodal to message, and also create errormessage component
export default class MessageModal extends React.PureComponent {
    render() {
        return (
            <div>
                {this.props.props.message}
            </div>
        )
    }
}
