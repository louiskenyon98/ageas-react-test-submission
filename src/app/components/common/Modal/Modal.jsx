import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../../themes/style.scss';

class Modal extends React.Component {
    render() {
        if (this.props.show) {
            return (
                <React.Fragment>
                    <div className={styles.veil} onClick={this.props.close}/>
                    <div className={styles.modal}>
                        {this.props.title && <h1>{this.props.title}</h1>}
                        {this.props.children}
                    </div>
                </React.Fragment>
            )
        }
        return null;
    }

}

Modal.propTypes = {
    show: PropTypes.bool,
    close: PropTypes.func,
    title: PropTypes.string,
    children: PropTypes.object
};

export default Modal;
