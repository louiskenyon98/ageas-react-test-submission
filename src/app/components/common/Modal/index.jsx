import React from 'react';
import styles from '../../../../themes/style.scss';

class Modal extends React.PureComponent {
    render() {
        const {show, children, closeModal} = this.props;
        //1 is placeholder for visible prop.
        if (show) {
            return (
                <React.Fragment>
                    <div className={styles.veil}/>
                    <div className={styles.modal}>
                        {children}
                        {/*<button className={styles.modalButton} onClick={closeModal}>close</button>*/}
                        <button className={`${styles.modalButton} ui inverted primary button`} onClick={closeModal}>close</button>
                    </div>
                </React.Fragment>
            )
        }
        return null;
    }

}

export default Modal;
