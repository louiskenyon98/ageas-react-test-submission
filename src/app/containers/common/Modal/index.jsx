import React from 'react';
import {connect} from 'react-redux';
import {closeModal} from '../../../actions/common/modal.action';
import Modal from '../../../components/common/Modal/Modal';
import StatusMessage from "../../../components/common/Modal/StatusMessage/Index";
import DeleteNoteContainer from "./ConfirmationMessage/DeleteNoteContainer";

export class ModalContainer extends React.Component {
    returnModal() {
        switch (this.props.modal.type) {
            case 'status':
                return (
                    <StatusMessage
                        {...this.props.modal.props}
                        close={this.props.closeModal}
                    />
                );
            case 'deleteNoteConfirmation':
                return (
                    <DeleteNoteContainer
                        {...this.props.modal.props}
                    />
                );
        }
    }

    render() {
        return (
            <Modal
                show={this.props.modal.show}
                close={this.props.closeModal}
            >
                {this.returnModal()}
            </Modal>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        modal: state.common.modal
    }
};

export default connect(mapStateToProps, {closeModal})(ModalContainer)
