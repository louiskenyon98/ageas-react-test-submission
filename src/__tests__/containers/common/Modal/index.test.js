import React from 'react';
import {shallow} from 'enzyme';
import {ModalContainer} from '../../../../app/containers/common/Modal';

describe('ModalContainer', () => {
    let props = {};
    let wrapper = () => shallow(<ModalContainer {...props}/>);
    beforeEach(() => {
        props = {
            modal: {
                type: 'status',
                props: {
                    foo: 'bar',
                },
                show: true,
            },
            closeModal: jest.fn()
        }
    });
    describe('render', () => {
        it('should render the StatusMessage component when type = status', () => {
            expect(wrapper()).toMatchSnapshot();
        });
        it('should render the ConfirmationMessage component when type = confirmation', () => {
            props.modal.type = 'confirmation';
            expect(wrapper()).toMatchSnapshot();
        });
        it('should set show prop to false when show = false', () => {
            props.modal.show = false;
            expect(wrapper()).toMatchSnapshot();
        });
    });
    describe('functionality', () => {
        describe('Modal', () => {
            it('should call prop closeModal on call of close prop', () => {
                expect(props.closeModal).not.toHaveBeenCalled();
                wrapper().find('Modal').first().props().close();
                expect(props.closeModal).toHaveBeenCalled();
            });
        })
    })
});
