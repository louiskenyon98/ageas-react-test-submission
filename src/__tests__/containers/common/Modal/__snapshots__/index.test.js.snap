// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalContainer render should render the Index component when type = confirmation 1`] = `
<Modal
  close={[MockFunction]}
  show={true}
/>
`;

exports[`ModalContainer render should render the StatusMessage component when type = status 1`] = `
<Modal
  close={[MockFunction]}
  show={true}
>
  <StatusMessage
    foo="bar"
  />
</Modal>
`;

exports[`ModalContainer render should set show prop to false when show = false 1`] = `
<Modal
  close={[MockFunction]}
  show={false}
>
  <StatusMessage
    foo="bar"
  />
</Modal>
`;
